File: quicksort-13b.rb
def quicksort(arr)
  return arr if arr.length <= 1
  pivot = arr[0]
  less = []
  greater = []
  arr.drop(1).each do |x|
    if x < pivot
      less << x
    else
      greater << x
    end
  end
  quicksort(less) + [pivot] + quicksort(greater)
end

File: quicksort-13b.js
function quicksort(arr) {
  if (arr.length <= 1) {
    return arr;
  }
  var pivot = arr[0];
  var less = [];
  var greater = [];
  for (var i = 1; i < arr.length; i++) {
    if (arr[i] < pivot) {
      less.push(arr[i]);
    } else {
      greater.push(arr[i]);
    }
  }
  return quicksort(less).concat([pivot], quicksort(greater));
}
